src/ft2font.cpp

```diff
@@ -361,6 +363,17 @@ void FT2Font::set_text(
         throw std::runtime_error("failed to set text flags for layout");
     }
 
+    if (languages.has_value()) {
+        for (auto & [lang_str, start, end] : languages.value()) {
+            if (!raqm_set_language(rq, lang_str.c_str(), start, end)) {
```

Comment:
This function actually takes a length, not an end location; I'm not sure which one we should accept, but I'll have to update this to the correct naming (the test is setup to match what libraqm says, but the docs say the opposite.)

Reply:
start-stop looks normal as API to me?